### PR TITLE
[DM] KSS for CSS examples

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- the first commit updates all the scss and md files in `core` (it's big and repetitive)
  - **scss files:** make them all valid KSS blocks again by adding header and reference.
references are now reflective of the CSS class being documented.
  - **md files:** add `@css reference` tags in all the CSS API sections
- update to documentalist@0.0.3, which includes a `KssPlugin` that does all the work 😮 
- write a theme handler for `@css reference` tags _(voila css examples are back)_
  - reuses the existing structure almost exactly, except references have changed quite significantly, which necessitated:
- ♻️  a major refactor to `_sections.scss` to support changes in how css examples are rendered
  - a new function `reference()` to target these new KSS examples. refactored as many selectors as I could, and removed a bunch of obsolete ones.

TODO

- [x] `pt-running-text` nesting issues
- [ ] syntax highlighting for the markup blocks (markup is missing until palantir/documentalist#16)
- [ ] markdown support in modifier docs (can anyone find broken examples?)

#### Reviewers should focus on:

- do all the CSS examples and modifier tables look good? any rough edges I missed?

#### Screenshot

image

### CODE REVIEW

File: packages/docs/src/styles/_sections.scss [hunk 1]
```diff
@@ -38,98 +50,133 @@
 
 // Section-Specific Styles
 
-#{section-id("components.button-group")} {
-  .docs-button-group-example-set .pt-button-group {
-    margin-top: $pt-grid-size * 2;
-    margin-left: $pt-grid-size * 2;
-  }
+#{reference("pt-file-upload")},
```
Comment (llorca) [inline packages/docs/src/styles/_sections.scss]: Cool mixin!

File: packages/core/src/components/breadcrumbs/breadcrumbs.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ containing breadcrumbs that are collapsed due to layout constraints.
 * When adding another element (such as a [tooltip](#components.tooltip) or
 [popover](#components.popover)) to a breadcrumb, wrap it around the contents of the `li`.
 
+@css pt-breadcrumbs
```
Comment (themadcreator) [inline packages/core/src/components/breadcrumbs/breadcrumbs.md]: 🔥

**Comment by themadcreator** [comment]
You can dismiss my review if you don't intend this PR to include packages other than core since it's a side branch PR.

**Comment by giladgray** [comment]
@themadcreator table + datetime are in #818 

@cmslewis i fixed the `.pt-running-text` nesting issues!

syntax highlighting + markdown require https://github.com/palantir/documentalist/pull/16

File: packages/docs/src/common/cssExample.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            {renderMarkupForModifier(markup, DEFAULT_MODIFIER)}
+            {modifiers.map((mod) => renderMarkupForModifier(markup, mod))}
+        </div>
+        <div className="kss-markup" dangerouslySetInnerHTML={{ __html: markupHtml }} />
+    </div>
+);
+
+function renderMarkupForModifier(markup: string, modifier: IKssModifier) {
+    const { name } = modifier;
+    const html = markup.replace(MODIFIER_PLACEHOLDER, (_, prefix) => {
```
Comment (giladgray) [inline packages/docs/src/common/cssExample.tsx]: copied from https://github.com/palantir/blueprint/blob/release-1.11.1/packages/docs/src/components/section.tsx#L160-L172

File: packages/docs/src/common/cssExample.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            return name;
+        } else {
+            return "";
+        }
+    });
+    return (
+        <div className="kss-example" data-modifier={modifier.name} key={modifier.name}>
+            <code>{modifier.name}</code>
+            <div dangerouslySetInnerHTML={{ __html: html }} />
+        </div>
```
Comment (giladgray) [inline packages/docs/src/common/cssExample.tsx]: https://github.com/palantir/blueprint/blob/release-1.11.1/packages/docs/src/components/section.tsx#L111-L114

File: packages/docs/src/common/cssExample.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export const CssExample: React.SFC<IKssExample> = ({ markup, markupHtml, modifiers, reference }) => (
+    <div>
+        {modifiers.length > 0 ? <ModifierTable modifiers={modifiers} /> : undefined}
+        <div className="kss-example-wrapper" data-reference={reference}>
+            {renderMarkupForModifier(markup, DEFAULT_MODIFIER)}
+            {modifiers.map((mod) => renderMarkupForModifier(markup, mod))}
+        </div>
+        <div className="kss-markup" dangerouslySetInnerHTML={{ __html: markupHtml }} />
```
Comment (giladgray) [inline packages/docs/src/common/cssExample.tsx]: https://github.com/palantir/blueprint/blob/release-1.11.1/packages/docs/src/components/section.tsx#L81-L87

File: packages/docs/src/common/propsStore.tsx [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
+import { ITsInterfaceEntry, ITsPropertyEntry } from "documentalist/dist/plugins/typescript";
 
-export interface IInheritedPropertyEntry extends IPropertyEntry {
+export interface IInheritedPropertyEntry extends ITsPropertyEntry {
     inheritedFrom?: string;
 }
 
 export class PropsStore {
-    constructor(private props: IInterfaceEntry[]) {}
+    constructor(private props: { [name: string]: ITsInterfaceEntry }) {}
```
Comment (giladgray) [inline packages/docs/src/common/propsStore.tsx]: https://github.com/palantir/documentalist/pull/12

**Comment by themadcreator** [comment]
Approved since other PR includes table

**Pull request merged**